reginaldCog/reginald.py

```diff
@@ -0,0 +1,180 @@
+import discord
+import json
+import openai
+import os
+import random
+import requests
+import base64
+import aiohttp
+from io import BytesIO
+from PIL import Image
+import tempfile
+from openai import OpenAIError
+from redbot.core import Config, commands
+
+
+class ReginaldCog(commands.Cog):
+    def __init__(self, bot):
+        self.bot = bot
+        self.config = Config.get_conf(self, identifier=71717171171717)
+        default_global = {
+            "openai_model": "gpt-3.5-turbo"
+        }
+        default_guild = {
+            "openai_api_key": None
+        }
+        self.config.register_global(**default_global)
+        self.config.register_guild(**default_guild)
+
+    def has_kanium_role():
+        async def predicate(ctx):
+            kanium_role_id = 280260875678515200
```

Comment:
I know reginald is kanium specific but i wouldve loved to see this more custom and not mention kanium in variable names. also role id hardcoded..